cucumber-selenium-tests/src/test/java/com/skillstorm/PageObjects/Components/Component.java

```diff
@@ -0,0 +1,13 @@
+/**
+ * By: Aaron Huggins
+ * 
+ * Description:
+ *      Serves as the base of the POM architecture and provides the common
+ *      functions for all components.
+ */
+
+package com.skillstorm.PageObjects.Components;
+
+public interface Component {
```

Comment:
Should this have a more specific name than "Component"?

Reply:
Mostly asking because Im not sure what this is supposed to do other than have a random "click button" method for now

Reply:
I'd like a more specific name as well, not entirely sure what to name it though. I named it "Component" after React Components.

Reply:
I see, can you explain what your idea is for the overarching concept of what this will do, that may give some more insight into naming it for its functionality... or is this just supposed to represent every possible component you can have in React?

Reply:
It's supposed to be the highest level representation in the POM, and is implemented by things like Navbar and Page. I suppose a good way to put it would be that it represents every possible component you can have in React that has child components you can interact with.

Reply:
Seems like a generic React Components file for the different interactions with web elements, I think we could keep it as generic as Component.java for now

Reply:
Since it is an interface though, this means that every object that represents a component MUST implement a method called "clickButton", but what if there is a component that doesn't need to click buttons, but wants to do something else... just seems too generic. Ill still approve it and all but it just doesn't make sense to me right now

Reply:
resolving thought the question still remains lol